# Python Series – Day 1: Introduction to Python

Welcome to Day 1 of the Python Series! In this notebook, we'll explore the fundamentals of Python programming, from setting up your environment to writing your first lines of code.

## 1. Introduction

### What is Python?
Python is a high-level, interpreted programming language known for its simple and readable syntax. It was created by Guido van Rossum and released in 1991. Python emphasizes code readability and allows developers to express concepts in fewer lines of code compared to languages like Java or C++.

### Why is Python Popular?
1. **Easy to Learn**: Python's syntax is intuitive and beginner-friendly
2. **Versatile**: Used across various domains (web, AI, data science, automation)
3. **Large Community**: Extensive libraries and frameworks available
4. **Active Development**: Constantly evolving with new features
5. **Open Source**: Free to use and modify

### Real-World Applications of Python
- **Artificial Intelligence & Machine Learning**: TensorFlow, PyTorch, scikit-learn
- **Web Development**: Django, Flask, FastAPI
- **Data Science & Analytics**: Pandas, NumPy, Matplotlib
- **Automation**: Scripting, web scraping, task automation
- **Game Development**: Pygame
- **Scientific Research**: NumPy, SciPy
- **Desktop Applications**: Tkinter, PyQt

## 2. Setting Up the Environment

### Installing Python

**Step 1: Download Python**
- Visit [python.org](https://www.python.org)
- Click on "Downloads" and select your operating system (Windows, macOS, or Linux)
- Download Python 3.9 or higher (recommended: Python 3.10+)

**Step 2: Install Python**
- Run the installer
- **Important**: Check the box "Add Python to PATH" during installation
- Click "Install Now"

**Step 3: Verify Installation**
Open Command Prompt (Windows) or Terminal (macOS/Linux) and type:
```
python --version
```
You should see your Python version displayed.

### What is Jupyter Notebook?
Jupyter Notebook is an open-source web application that allows you to create and share documents containing live code, equations, visualizations, and narrative text. It supports multiple programming languages, with Python being the most popular.

**Key Features:**
- Write and execute code in cells
- Mix code with markdown documentation
- Visualize data with graphs and charts
- Save and share notebooks easily

### How to Run Cells

1. **Running a Single Cell**: Click on a cell and press `Shift + Enter`
2. **Running All Cells**: Use the menu `Cell → Run All`
3. **Clear Output**: `Cell → All Output → Clear`
4. **Cell Types**:
   - **Code Cell**: Contains executable Python code
   - **Markdown Cell**: Contains formatted text, headings, and documentation

## 3. First Python Code

Let's write our first Python program! We'll explore the print() function, comments, and basic variable assignment.

In [1]:
# Our first Python program - Hello, Python!
# https://colab.research.google.com/

print('Hello World')

Hello World


### Understanding Comments and Variables

**Comments** are lines of text that Python ignores. They help explain your code:
- Single-line comments use `#`
- Multi-line comments use `'''` or `"""`

**Variables** are containers for storing data values. Python automatically determines the data type.

In [4]:
# This is a comment - Python ignores this line
# Variables store data values

# Creating variables
name = 'Hello'



# Printing variables
print(name)

# Multiple prints in one line
print(name, 10, True) # Hello 10 True

Hello
Hello 10 True


## 4. Basic Data Types

Python has several built-in data types. Let's explore the most common ones:

### Integer (int)
Whole numbers without decimal points. They can be positive or negative.

In [2]:
# Integer examples

num = 10
print(num)
print(type(num))


# Basic arithmetic with integers
10 + 1

10
<class 'int'>


11

### Float (float)
Numbers with decimal points. Used for more precise values.

In [5]:
# Float examples
num = 10.0 
print(type(num))

# Operations with floats

10.0 - 1

<class 'float'>


9.0

### String (str)
Text data enclosed in single quotes '', double quotes "", or triple quotes '''.

In [15]:
# String examples
char = 'Khushi'
char1 = "ABC"

char2 = '''Shyam is good
shyam is also good
''' # This is used for multi-line strings


print(char)
print(char1)
print(char2)


# String concatenation
print(char+char1) # khushiABC


# String repetition * 3
print(char1*2)
'HIII' * 10

Khushi
ABC
Shyam is good
shyam is also good

KhushiABC
ABCABC


'HIIIHIIIHIIIHIIIHIIIHIIIHIIIHIIIHIIIHIII'

### Boolean (bool)
A data type with only two values: True or False. Used for logical operations.

In [6]:
# Boolean examples
is_flag = True # 10 = 10
print(type(is_flag))

is_flag1 = False

# Boolean from comparisons 
# == Equal to
# != Not eqaul to
# < less tahn
# > greater than

print(is_flag == is_flag1) # False

is_flag1 != is_flag # True

<class 'bool'>
False


True

## 5. Input & Output

### Taking Input from Users

The `input()` function allows you to get data from the user. It always returns a string.

In [2]:
name = input()

print(type(name), name)

<class 'str'> 10


### Type Conversion

Since `input()` returns a string, we need to convert it to other data types if necessary.

In [4]:
# Type conversion examples
# num = int(input())
# print(type(num), num)

# Convert to string

num1 = 10

num1 = str(num1)
type(num1)

# Simulating user input and calculations
user_input_age = "30"  
age = int(user_input_age)
birth_year = 2025 - age
print(f"You were born approximately in {birth_year}")

You were born approximately in 1995


## 6. Simple Exercises

### Exercise 1: Personal Information

Create three variables for your personal information and print them in a formatted sentence.

In [9]:
# Exercise 1: Personal Information
# Create and display personal information

name = "Khushi"
age = 80
hobby = "Watching Serials"

# Method 1: Using print with commas
print("Method 1:")
print(name, age, hobby)

# Method 2: Using concatenation
print("\nMethod 2:")
print("My name is " + name + " I am " + str(age) + " years old, and I love " + hobby)

# Method 3: Using f-strings (Recommended - most modern and readable)
print("\nMethod 3 (Best):")
print(f"My name is {name}, I am {age} years old, and I love {hobby}.")

Method 1:
Khushi 80 Watching Serials

Method 2:
My name is Khushi I am 80 years old, and I love Watching Serials

Method 3 (Best):
My name is Khushi, I am 80 years old, and I love Watching Serials.


In [7]:
print("Harsh\nKhushi")

Harsh
Khushi


In [ ]:
# Pydantic is a Python library that uses type hints for data validation and parsing. It ensures that data is clean, correctly structured, and conforms to defined rules before being processed, which makes applications more robust and reliable. Pydantic handles this by defining data models with type annotations, automatically converting and validating data, and providing clear error messages if validation fails

### Exercise 2: Temperature Conversion

Convert temperature from Celsius to Fahrenheit using the formula:
$$F = (C \times 9/5) + 32$$

In [ ]:
# Exercise 2: Temperature Conversion
# Convert Celsius to Fahrenheit

celsius = 25
fahrenheit = (celsius * 9/5) + 32

print(f"{celsius}°C = {fahrenheit}°F") # 30°C

# # Try with different temperatures
temps_celsius = [0, 10, 20, 30, 100, -40]
print("\nTemperature Conversion Table:")
print("Celsius → Fahrenheit")
print("-" * 25)

for temp_c in temps_celsius:
    temp_f = (temp_c * 9/5) + 32
    print(f"{temp_c:>5}°C → {temp_f:>7.1f}°F") # temp_f:>7.1f --- iska mtlb decimal 1 se jyda nahi hona chahiye

25°C = 77.0°F

Temperature Conversion Table:
Celsius → Fahrenheit
-------------------------
    0°C →    32.0°F
   10°C →    50.0°F
   20°C →    68.0°F
   30°C →    86.0°F
  100°C →   212.0°F
  -40°C →   -40.0°F


### Challenge: Create Your Own Exercise

Now, try to create your own simple program! Here are some ideas:
- Create a program that calculates the area of a circle given a radius
- Calculate the average of three test scores
- Convert kilometers to miles (1 km = 0.621371 miles)
- Calculate the total cost of items with tax (tax rate: 5%)

**Try one of these challenges below:**

In [15]:
# Challenge Solution Example: Calculate the area of a circle
import math

raduis = int(input())

pi = math.pi
print(pi)

# Area of cirlce is πr2
print(f"Area of circle with {raduis} is {pi * (raduis**2)}")


# Another example: Average of test scores

3.141592653589793
Area of circle with 7 is 153.93804002589985


## 7. Day 1 Summary

### Key Points Learned

1. **Introduction to Python**
   - Python is a versatile, beginner-friendly language
   - Used in AI, ML, Web Development, Data Science, and Automation

2. **Environment Setup**
   - Install Python from python.org
   - Use Jupyter Notebook for interactive programming
   - Run cells with Shift + Enter

3. **Basic Syntax**
   - Use `print()` to display output
   - Comments start with `#`
   - Variables store data values

4. **Data Types**
   - **int**: Whole numbers (e.g., 42, -5)
   - **float**: Decimal numbers (e.g., 3.14, 19.99)
   - **str**: Text (e.g., "Hello", 'Python')
   - **bool**: True or False

5. **Input & Output**
   - `input()` gets data from users
   - Use `int()`, `float()`, `str()` to convert types
   - Use f-strings for formatted output: `f"Value: {variable}"`

6. **Practice**
   - Created personal information display
   - Performed temperature conversion
   - Solved mathematical problems

### What's Next? Day 2: Operators & Expressions

On Day 2, we'll explore:
- Arithmetic operators (+, -, *, /, //, %, **)
- Comparison operators (==, !=, <, >, <=, >=)
- Logical operators (and, or, not)
- Order of operations
- Complex expressions

### Practice Tasks for You

Before moving to Day 2, try these:
1. Write a program to calculate the BMI (Body Mass Index)
   - Formula: BMI = weight (kg) / (height (m))²
2. Create a program that converts currency (e.g., USD to EUR)
3. Calculate compound interest using: A = P(1 + r/n)^(nt)

**Keep practicing and happy coding!** 🚀